### PR TITLE
CAMEL-23638: camel-jbang - Add CLI meta-tools to camel ask

### PR DESCRIPTION
**Pull request by gnodet**
[CAMEL-23638](https://issues.apache.org/jira/browse/CAMEL-23638)

## Summary

Currently `camel ask` has ~26 hardcoded tool definitions. New CLI features (like `camel get error --diagram`) are not automatically available to the LLM. This adds 3 CLI meta-tools that give the LLM progressive access to the entire 126-command CLI:

- **`cli_list_commands(filter?)`** — discover available commands from the auto-generated `camel-jbang-commands-metadata.json`
- **`cli_command_help(command)`** — get detailed help for a specific command
- **`cli_exec(command)`** — execute any CLI command in-process and return captured output

The existing curated tools are preserved for common operations. The meta-tools serve as an escape hatch for everything else.

### How the LLM uses it

Example flow for `camel get error --diagram`:
1. User asks "why did my message fail?"
2. LLM calls `cli_list_commands("error")` → sees `get error: Get captured routing errors...`
3. LLM calls `cli_command_help("get error")` → sees `--diagram`, `--detail`, `--last`, etc.
4. LLM calls `cli_exec("get error --diagram")` → gets the diagram + error details

### Design decisions

- Metadata loaded from the auto-generated JSON already on the classpath — cached after first read
- Execution via `CamelJBangMain.getCommandLine().execute()` in-process (no subprocess)
- Output captured by temporarily swapping both `Printer` and picocli `PrintWriter` (both restored in `finally`)
- Command tokenization handles quoted strings (`--filter="my route"`)
- New CLI commands are automatically available without code changes to Ask.java

## Test plan

- [ ] Verify `cli_list_commands` returns all commands from metadata
- [ ] Verify `cli_list_commands` with filter narrows results correctly
- [ ] Verify `cli_command_help("get error")` returns all options
- [ ] Verify `cli_exec("catalog component --filter=kafka")` returns output
- [ ] Verify existing curated tools still work unchanged
- [x] Unit tests for command tokenization (quoted strings, extra spaces, edge cases)

_Claude Code on behalf of Guillaume Nodet_

### CODE REVIEW

**Review by davsclaus — APPROVED** [review]

File: dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java [hunk 1]
```diff
@@ -720,6 +749,213 @@ private String executeGetExampleFile(JsonObject args) {
...
+
+    @SuppressWarnings("unchecked")
+    private List<JsonObject> loadCommandMetadata() {
+        try (InputStream is = getClass().getClassLoader()
+                .getResourceAsStream("META-INF/camel-jbang-commands-metadata.json")) {
+            if (is == null) {
+                return List.of();
+            }
+            String json = IOHelper.loadText(is);
+            JsonObject root = (JsonObject) Jsoner.deserialize(json);
```
Comment (davsclaus) [inline dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java:766]: Minor: `loadCommandMetadata()` is called fresh on each tool invocation. Since the metadata JSON is a static classpath resource, consider caching the parsed result in a field (e.g., `private List<JsonObject> cachedCommands;`) to avoid repeated I/O and parsing during a chat session.

File: dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java [hunk 2]
```diff
@@ -720,6 +749,213 @@ private String executeGetExampleFile(JsonObject args) {
...
+        picocli.CommandLine commandLine = CamelJBangMain.getCommandLine();
+        if (commandLine == null) {
+            return "Error: CLI not available";
+        }
+
+        String[] cmdArgs = command.trim().split("\\s+");
+
+        // capture output by temporarily swapping the Printer on main
+        StringBuilder captured = new StringBuilder();
+        Printer capturingPrinter = new Printer() {
```
Comment (davsclaus) [inline dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java:916]: Question: `command.trim().split("\\s+")` won't handle quoted arguments correctly (e.g., `get route --filter="my route"`). Is this a known limitation? If so, it might be worth noting in the tool description so the LLM avoids passing values with spaces.

File: dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java [hunk 3]
```diff
@@ -720,6 +749,213 @@ private String executeGetExampleFile(JsonObject args) {
...
+            String output = captured.toString() + sw.toString();
+            if (output.isBlank() && exitCode != 0) {
+                return "Command exited with code " + exitCode;
+            }
+            if (output.length() > 32768) {
+                output = output.substring(0, 32768) + "\n... (truncated)";
+            }
+            return output;
+        } catch (Exception e) {
+            return "Error executing command: " + e.getMessage();
```
Comment (davsclaus) [inline dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java:960]: Minor: The `finally` block restores `getMain().setOut(originalPrinter)` but doesn't restore picocli's `commandLine.setOut()` / `setErr()` to their original values. If an exception occurs between setting them (lines 934-935) and the `finally`, the writers stay swapped for subsequent calls.

**Review by davsclaus — APPROVED** [review]
Nice addition — the three CLI meta-tools are a clean way to give the LLM progressive access to the full CLI without bloating the curated tool list. A few minor observations and questions:

1. **Tests**: No tests for the new metadata loading and command discovery logic (`loadCommandMetadata`, `collectCommands`, `findCommand`). These are easily unit-testable without an LLM and would catch regressions.

2. **Metadata caching**: `loadCommandMetadata()` re-reads and re-parses the classpath JSON on every tool call. Since it's static, caching it in a field would save repeated I/O in a chat session.

3. **Question — quoted arguments**: `cli_exec` splits on `\s+`, so `get route --filter="my route"` won't work as expected. Worth documenting in the tool description?

4. **Question — picocli out/err restore**: The `finally` block restores the `Printer` but not the `commandLine.setOut()`/`setErr()` writers. If an exception occurs, they stay swapped.

Overall the design is solid — in-process picocli execution, output truncation at 32KB, and the system prompt update for LLM discoverability are all good choices. Approving.

This review is a project-rules and conventions check. It does not replace specialized review tools such as CodeRabbit or SonarCloud.

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._

_Claude Code on behalf of Claus Ibsen_

File: dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java (hunk 1 above)
Comment (gnodet) [reply dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java:766]: Fixed — metadata is now cached in a `commandMetadataCache` field after first load.

_Claude Code on behalf of Guillaume Nodet_

File: dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java (hunk 2 above)
Comment (gnodet) [reply dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java:916]: Fixed — replaced `split("\\s+")` with a proper `tokenizeCommand()` that handles single and double quoted strings. Added unit tests covering this.

_Claude Code on behalf of Guillaume Nodet_

File: dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java (hunk 3 above)
Comment (gnodet) [reply dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/Ask.java:960]: Fixed — `commandLine.setOut(originalOut)` and `commandLine.setErr(originalErr)` are now restored in the `finally` block alongside the Printer.

_Claude Code on behalf of Guillaume Nodet_

**Comment by davsclaus** [comment]
LGTM
